src/cursor/cur_hs.c

```diff
@@ -883,6 +883,12 @@ __curhs_insert(WT_CURSOR *cursor)
         goto retry;
     WT_ERR(ret);
 
+    /*
+     * Mark the insert as successful. Even if one of the calls below fails, some callers will still
+     * need to know whether the actual insert went through or not.
+     */
+    F_SET(hs_cursor, WT_HS_CUR_INSERT_SUCCESS);
```

Comment:
The issue is that anything below this point can fail and make the insert return value non-zero despite the fact that we've successfully inserted into the table.